transforms/universal/html2parquet/python/src/html2parquet_main.py

```diff
@@ -0,0 +1,110 @@
+        """
+        Converts raw data file (ZIP) to Parquet format
+        """
+        # We currently only process .zip files
+        print("transform_binary")
+        if TransformUtils.get_file_extension(file_name)[1] != ".zip":
+            self.logger.warning(f"Got unsupported file type {file_name}, skipping")
+            return [], {}
+        data = []
+        number_of_rows = 0
+
+        with zipfile.ZipFile(io.BytesIO(bytes(byte_array))) as opened_zip:
+            # Loop through each file member in the ZIP archive
+            for member in opened_zip.infolist():
+                if not member.is_dir():
+                    with opened_zip.open(member) as file:
+                        try:
+                            # Read the content of the file
+                            content_bytes = file.read()
+                            #use Tra something library
+                            content_string = trafilatura.extract(content_bytes)
+                            print("content_string:", content_string)
+                            row_data = {
+                                "filename": member.filename,
+                                "contents": content_string
+                            }
+
+                            data.append(row_data)
+                            number_of_rows += 1
+                        except Exception as e:
+                            self.logger.warning(f"Exception {str(e)} processing file {member.filename}, skipping")
+        table = pa.Table.from_pylist(data)
+        return [(TransformUtils.convert_arrow_to_binary(table=table), ".parquet")], {"number of rows": number_of_rows}
+
+
+ 
+    def transform(self, table: pa.Table, file_name: str = None) -> tuple[list[pa.Table], dict[str, Any]]:
+        print("transform")
+        if self.sleep is not None:
+            time.sleep(self.sleep)
+        # Add some sample metadata.
+        metadata = {"nfiles": 1, "nrows": len(table)}
+        return [table], metadata
+
+
+short_name = "html2parquet"
+cli_prefix = f"{short_name}_"
+sleep_key = "sleep_sec"
+pwd_key = "pwd"
+sleep_cli_param = f"{cli_prefix}{sleep_key}"
+pwd_cli_param = f"{cli_prefix}{pwd_key}"
+
+
+class HtmlToParquetTransformConfiguration(TransformConfiguration):
+    def __init__(self):
+        super().__init__(
+            name=short_name,
+            transform_class=HtmlToParquetTransform,
+            remove_from_metadata=[pwd_key],
+        )
+    def add_input_params(self, parser: ArgumentParser) -> None:
+        parser.add_argument(
+            f"--{sleep_cli_param}",
+            type=int,
+            default=1,
+            help="Sleep actor for a number of seconds while processing the data frame, before writing the file to COS",
+        )
+        parser.add_argument(
+            f"--{pwd_cli_param}",
+            type=str,
+            default="nothing",
+            help="A dummy password which should be filtered out of the metadata",
+        )
```

Comment:
The above is from Noop and does not belong here. Pls add only the parameters that you need